feat: adds --watch to introspect commands (#1207)
apollographql/apollo-cli: docs/source/commands/subgraphs.mdx

> Unlike a standard introspection query, the result of `rover subgraph introspect` _does_ include certain directives (specifically, directives related to federation like `@key`). This is possible because the command uses a separate introspection mechanism provided by the [Apollo Federation specification](/federation/federation-spec/#fetch-service-capabilities).

> #### Using the `--watch` flag

Reviewer: ```suggestion
#### Watching for schema changes
```

> You can pass `--watch` to `rover subgraph introspect` and Rover will introspect your subgraph every second. Rover will only print an updated introspection response (or error message) to your terminal if the output of the query differs from the output of the previous query. 

Reviewer: ```suggestion
If you pass `--watch` to `rover subgraph introspect`, Rover introspects your subgraph every second. Whenever the returned schema differs from the _previously_ returned schema, Rover outputs the updated schema.  
```
